Tests/Amr/Advection_AmrLevel/Source/AmrLevelAdv.cpp

```diff
@@ -710,19 +710,26 @@ AmrLevelAdv::errorEst (TagBoxArray& tags,
 {
     MultiFab& S_new = get_new_data(Phi_Type);
 
+    // Properly fill patches and ghost cells for phi gradient check.
+    MultiFab phi(S_new.boxArray(), S_new.DistributionMap(), NUM_STATE, 1);
+    if (level < max_phigrad_lev) {
+        const Real cur_time = state[Phi_Type].curTime();
+        FillPatch(*this, phi, 1, cur_time, Phi_Type, 0, NUM_STATE);
+    }
+
```

Comment:
I think you should do something like this to avoid allocating a temporary MultiFab when level is not less than max_phigrad_lev. 
```
MultiFab phitmp;
if (level < max_phigrad_lev) {
    phitmp.define(....);
    FillPatch(....)
}

MultiFab const& phi = (level < max_phigrad_lev) ? phitmp : S_new;
```

Then use phi in both tagging functions in the code following this.

Reply:
I like that approach, uses less memory. Lemme commit it real quick.